src/ui/Facet/Facet.ts

```diff
@@ -1517,8 +1517,15 @@ export class Facet extends Component {
     }
   }
 
-  private initDependentFacetManager() {
-    this.dependentFacetManager = new DependentFacetManager(this);
+  private initDependsOnManager() {
+    const facetInfo: IDependentFacet = {
```

Comment:
Building an new `IDependentFacet` object offers better encapsulation and flexibility with naming, but in this case all the passed properties and methods are already `public`. Would it not have been shorter to do:

```
export class Facet extends Component implements IDependentFacet {}
```

and then:
```
this.dependsOnManager = new DependsOnManager(this);
```

The small downside is you would need to add a public getter:
```
public get dependsOn() { return this.options.dependsOn; }
```
but I think this is okay because because we are saying a Facet should "look" like an IDependentFacet.

What do you think @olamothe?

Reply:
After talking with @olamothe, I will keep my current implementation.

Here's why:

My interface needs a function `reset`. It already exists on both the component, but I can't use it for the `categoryFacet` because it launches a new query. This cause an error because a query was already executing from the master facet selection changing. 

So, if I was to implement the interface directly on the components, I would need to create a new reset function on both components and only use them in the `dependsOnManager`